src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/SNSService.kt

```diff
@@ -78,6 +80,36 @@ class SNSReferralService(
         )
         snsPublisher.publish(event.referral.id, assignment.assignedBy, snsEvent)
       }
+      ReferralEventType.CANCELLED -> {
+        val snsEvent = EventDTO(
+          "intervention.referral.cancelled",
+          "A referral has been cancelled",
+          event.detailUrl,
+          event.referral.concludedAt!!,
+          mapOf("referralId" to event.referral.id)
+        )
+        snsPublisher.publish(event.referral.id, event.referral.endRequestedBy!!, snsEvent)
+      }
+      ReferralEventType.PREMATURELY_ENDED -> {
+        val snsEvent = EventDTO(
+          "intervention.referral.prematurely-ended",
+          "A referral has been ended prematurely",
+          event.detailUrl,
+          event.referral.concludedAt!!,
+          mapOf("referralId" to event.referral.id)
+        )
+        snsPublisher.publish(event.referral.id, event.referral.endRequestedBy!!, snsEvent)
+      }
+      ReferralEventType.COMPLETED -> {
+        val snsEvent = EventDTO(
+          "intervention.referral.completed",
+          "A referral has been completed",
+          event.detailUrl,
+          event.referral.concludedAt!!,
+          mapOf("referralId" to event.referral.id)
+        )
+        snsPublisher.publish(event.referral.id, event.referral.currentAssignee!!, snsEvent)
```

Comment:
I’m wondering if endOfServiceReport.submittedBy would be more suitable even though it's always same person (I think)

Reply:
Having thought about this, I think the person completing the referral is the assigned caseworker logically speaking. The person submitting the end of service report happens to be that person today (which automatically initiates the completion) but the person responsible for the referral from an SP perspective is the assigned person. Therefore, I think it's ok to leave it as it is.